### PR TITLE
Examples share a common gradle build script

### LINKED ISSUE #1041: Examples should share a common gradle build script

### PR DESCRIPTION
**Pull request by thisthat**
This resolves #1041 introducing a common Gradle structure for the examples. 
Now all examples use the same OTel version and have to adhere to the Google Java format.

### CODE REVIEW

File: examples/build.gradle [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+subprojects {
+    apply plugin: 'eclipse'
+    apply plugin: 'java'
+    apply plugin: 'java-library'
+    apply plugin: 'idea'
+    apply plugin: 'com.github.sherter.google-java-format'
+
+    group = "io.opentelemetry"
+    version = "0.4.0-SNAPSHOT" // CURRENT_OPEN_TELEMETRY_VERSION
```
Comment (bogdandrutu) [inline examples/build.gradle:14]: keep it as 0.1.0-SNAPSHOT, we don't release this to maven central so no need to increase the version or keep it in sync with the main repo.

File: examples/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
```
Comment (bogdandrutu) [inline examples/gradle/wrapper/gradle-wrapper.properties:1]: I expect more wrappers to be deleted from the sub-projects, did you forget to do that?

File: examples/grpc/build.gradle [hunk 3]
```diff
@@ -1,36 +1,21 @@
...
-group 'io.opentelemetry'
-version '0.1.0-SNAPSHOT'
-
-def opentelemetryVersion = '0.3.0'
-
+description = 'OpenTelemetry Examples for gRPC'
+ext.moduleName = "io.opentelemetry.examples.grpc"
 
 def grpcVersion = '1.24.0' // CURRENT_GRPC_VERSION
 def protobufVersion = '3.10.0'
```
Comment (bogdandrutu) [inline examples/grpc/build.gradle:9]: Move these constants to the main `build.gradle`.

File: examples/build.gradle [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        baseName = project.name + '-all'
+        from { configurations.compile.collect { it.isDirectory() ? it : zipTree(it) } }
+        with jar
+    }
+}
+
+wrapper {
+    distributionType = Wrapper.DistributionType.ALL
+    gradleVersion = '6.0.1'
+}
```
Comment (bogdandrutu) [inline examples/build.gradle:45]: don't think you need this.

File: examples/gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.0.1-all.zip
```
Comment (bogdandrutu) [inline examples/gradle/wrapper/gradle-wrapper.properties:3]: we use `6.3.0` now.

**Comment by carlosalberto** [comment]
Overall LGTM - once that Bogdan's issues are solved. Thanks for looking into this!

**Pull request merged**